apps/authoring-services/src/main/java/com/infosys/lexauthoringservices/serviceimpl/CompetencyServiceImp.java

```diff
@@ -0,0 +1,146 @@
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class CompetencyServiceImp implements CompetencyService {
+
+    @Autowired
+    private LexLogger logger;
+
+    @Autowired
+    private Producer producer;
+
+    @Autowired
+    private ContentCrudService contentCrudService;
+
+    @Autowired
+    private GraphServiceImpl graphService;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Value("${kafka.topics.competency.update}")
+    private String competencyUpdateTopic;
+
+    @Value("${sbext.service.host}")
+    private String sbextServiceURL;
+
+    @Value("${sbext.service.competencySearchPath}")
+    private String competencySearchPath;
+
+    public final String COMPETENCIES_CONST = "competencies";
+
+    public final String ID_CONST = "id";
+
+    public final String NAME_CONST = "name";
+
+    public final String DESCRIPTION_CONST = "description";
+
+    public final String IDENTIFIER_CONST = "identifier";
+
+    public final String TYPE_CONST = "type";
+
+    @Override
+    public void updateCompetencyData(Competency competency) {
+        producer.push(competencyUpdateTopic, competency);
+    }
+
+    /**
+     * Update the copetency in neo4j & elasticsearch
+     *
+     * @param competency
+     * @return
+     */
+    @Override
+    public Competency processUpdateCompetencyData(Competency competency) {
+        StringBuilder builder = new StringBuilder();
+        String path = competencySearchPath.replace("{org}", competency.getRootOrg()).replace("{id}", competency.getId());
+        builder.append(sbextServiceURL).append(path);
+        Object response = fetchResult(builder);
+        if (ObjectUtils.isEmpty(response))
+            return competency;
+        List<HashMap<String, Object>> listOfIds = (List<HashMap<String, Object>>) response;
+        if (!listOfIds.isEmpty()) {
+            List<String> successRecords = new ArrayList<>();
+            List<String> failedRecords = new ArrayList<>();
+            logger.info("Competency update started ..............");
+            for (HashMap<String, Object> map : listOfIds) {
+                String lex_id = (String) map.get(IDENTIFIER_CONST);
+                try {
+                    Map<String, Object> contentData = contentCrudService.getContentNode(competency.getRootOrg(), lex_id);
+                    Object competencyObject = contentData.get(COMPETENCIES_CONST);
+                    if (!ObjectUtils.isEmpty(competencyObject)) {
+                        List<HashMap<String, String>> oldCompetencyData = (List<HashMap<String, String>>) competencyObject;
+                        List<HashMap<String, String>> newCompetencyData = new ArrayList<>();
+                        HashMap<String, String> competencyMap = new HashMap<>();
+                        for (HashMap<String, String> data : oldCompetencyData) {
```

Comment:
This logic is error prone... tomorrow if we decided to add more fields, then we have to keep updating this code... 

We should update "name" and "description" in "data" object (which is from oldCompetencyData) -- and save oldCompetencyData -- this way we are not going to loose any other information in oldCompetencyData...

I suggest to get a list of keys from properties to replace - NAME_CONST, DESCRIPTION_CONST... read these as list from properties and check the "data" has these keys -- if so, update them... This way we can simply update this property list tomorrow if one more field is getting updated in the competency object